Mathlib/Data/Real/EReal.lean

```diff
@@ -1053,6 +1053,19 @@ lemma add_sub_cancel_right {a : EReal} {b : Real} : a + b - b = a := by
   · norm_cast; linarith
   · rw [top_add_of_ne_bot (coe_ne_bot b), top_sub_coe]
 
+lemma _root_.ENNReal.toEReal_sub {x y : ℝ≥0∞} (hy_top : y ≠ ∞) (h_le : y ≤ x) :
+    (x - y).toEReal = x.toEReal - y.toEReal := by
+  by_cases hx_top : x = ∞
+  · lift y to ℝ≥0 using hy_top
+    simp only [hx_top, top_sub_coe, EReal.coe_ennreal_top]
+    norm_cast
+  have h_top : x - y ≠ ∞ := by
+    simp only [ne_eq, sub_eq_top_iff, hx_top, hy_top, not_false_eq_true, and_true]
+  nth_rw 2 [← ofReal_toReal_eq_iff.mpr hy_top, ← ofReal_toReal_eq_iff.mpr hx_top]
```

Comment:
Can this line and the next somehow be golfed? This *feels* redundant to me. (Perhaps the answer is "no, a better tactic is needed", but it would be nice to think about it.)

Reply:
I'll try to golf it now.